pkg/diagnose/check.go

```diff
@@ -24,10 +26,92 @@ func init() {
 	diagnosis.Register("check-datadog", diagnose)
 }
 
+func diagnose(diagCfg diagnosis.Config) []diagnosis.Diagnosis {
+	if diagCfg.RunningInAgentProcess {
+		return diagnoseInAgentProcess(diagCfg)
+	}
+
+	return diagnoseInCLIProcess(diagCfg)
+}
+
+func partitionChecks(checks []check.Check) [][]check.Check {
+	instancesMap := map[string][]check.Check{}
+
+	// partition per check (instances of a check groupped by check/integration)
+	for _, c := range checks {
+		checkName := c.String()
+		instances, exists := instancesMap[checkName]
+		if exists {
+			instances = append(instances, c)
+		} else {
+			instances = make([]check.Check, 1)
+			instances[0] = c
+		}
+		instancesMap[checkName] = instances
```

Comment:
This is all redundant, this is is the same:
```
instancesMap[checkName] = append(instancesMap[checkName], c)
```

Reply:
Indeed, probably could make it better (still learning Golang ropes). Regarding code review of this section it becomes unnecessary since whole function is removed now, thank you.